presto-parser/src/main/java/com/facebook/presto/sql/tree/Window.java

```diff
@@ -13,99 +13,21 @@
  */
 package com.facebook.presto.sql.tree;
 
-import com.google.common.collect.ImmutableList;
-
-import java.util.List;
-import java.util.Objects;
 import java.util.Optional;
 
-import static com.google.common.base.MoreObjects.toStringHelper;
-import static java.util.Objects.requireNonNull;
-
-public class Window
+public abstract class Window
```

Comment:
As per the specs, the syntax for window function is defined as:
```<window function> ::= <window function type> OVER <window name or specification>```. So, for the parser object `FunctionCall`, we need a class member of type ```<window name or specification>```. Having an abstract `Window` class, from which `WindowName` or `WindowDefinition` are derived, makes it easier to represent this member by a single type, as is the case currently:
```
public class FunctionCall
        extends Expression
{
   ....
    private final Optional<Window> window;
```